### PR TITLE
[fix] Make dataset deduplication independent of Python's hash seed

### PR DESCRIPTION
**Pull request by pharmpy-dev-123**
This implementation takes data, dtypes, columns, and index into account for hash computation.

### CODE REVIEW

File: src/pharmpy/internals/df.py [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+
+
+def _df_hash_values(df: pd.DataFrame) -> Iterator[pd.Series]:
+    yield _pd_hash_values(df.columns)
+    yield _pd_hash_values(df.index)
+    yield _pd_hash_values(df.dtypes)
+    yield _pd_hash_values(df)
+
+
+def hash_df_runtime(df: pd.DataFrame) -> int:
```
Comment (stellabelin) [inline src/pharmpy/internals/df.py:23]: Is this used anywhere?

**Review by stellabelin — COMMENTED** [review]
LGTM!

File: src/pharmpy/internals/df.py (hunk 1 above)
Comment (pharmpy-dev-123) [reply src/pharmpy/internals/df.py:23]: No, but I think it could be used if we ever need to put dataframes in sets or dicts.

**Review by rikardn — APPROVED** [review]
LGTM

**Pull request merged**